codalab/worker/bundle_manager.py

```diff
@@ -529,3 +533,20 @@ def _construct_run_message(self, worker, bundle):
         resources['request_network'] = bundle.metadata.request_network
 
         return message
+
+    def _bring_zombie_bundles_to_fail(self):
+        failure_message = "Bundle exceeds maximum time limit {}.".format(BUNDLE_TIMEOUT_SECONDS)
+        bundles_to_fail = (
+            self._model.batch_get_bundles(state=State.UPLOADING)
+            + self._model.batch_get_bundles(state=State.STAGED)
+            + self._model.batch_get_bundles(state=State.RUNNING)
+        )
+        now = time.time()
+
+        for bundle in bundles_to_fail:
+            if now - bundle.metadata.created > BUNDLE_TIMEOUT_SECONDS:
```

Comment:
This is not ideal because a bundle could have waited a while until it got staged.  Is there a last updated field for runs?  If not, then just put a comment here and leave it as is.

Reply:
I only found `last_updated` in bundles with bundle_type == run. I didn't find this field in other types of bundles. So maybe a better way would be checking on different field based on bundle_type?